lib/src/pub.dart

```diff
@@ -5,25 +5,40 @@
 import 'dart:convert';
 import 'dart:io';
 
+import 'package:cli_util/cli_util.dart';
 import 'package:grinder/grinder.dart';
 import 'package:path/path.dart' as p;
+import 'package:pub_semver/pub_semver.dart';
 
 import 'config_variable.dart';
 import 'utils.dart';
 
-/// The entire contents of pub's `~/.pub-cache/credentials.json` file.
+/// The entire contents of the `pub-credentials.json` file.
 ///
 /// **Do not check this in directly.** This should only come from secure
 /// sources.
 ///
+/// You can find your current `pub-credentials.json` in the following locations:
+///
+/// * Dart prior to 2.15.0: `$HOME/.pub-cache/credentials.json`.
+/// * Linux: `$XDG_CONFIG_HOME/dart/pub-credentials.json` if `$XDG_CONFIG_HOME`
+///   is defined, otherwise `$HOME/.config/dart/pub-credentials.json`.
+/// * Mac OS: `$HOME/Library/Application Support/dart/pub-credentials.json`
+/// * Windows: `%APPDATA%/dart/pub-credentials.json`
+///
 /// By default this comes from the `PUB_CREDENTIALS` environment variable.
 final pubCredentials = InternalConfigVariable.fn<String>(() =>
     Platform.environment["PUB_CREDENTIALS"] ??
     fail("pkg.pubCredentials must be set to deploy to pub."));
 
 /// The path in which pub expects to find its credentials file.
 final String _credentialsPath = () {
-  // This follows the same logic as pub:
+  // This follows the same logic as pub.
+  if (dartVersion.major >= Version.parse('2.15.0')) {
```

Comment:
It looks like `major` returns an int so it can't be compared with `Version`. Did you mean `darVersion >= Version.parse('2.15.0')`?